File: server/scripts/initAdmin.js
const mongoose = require('mongoose');
const User = require('../models/User');

const MONGODB_URI = process.env.MONGODB_URI || 'mongodb://localhost:27017/any-go';

async function initAdmin() {
  try {
    await mongoose.connect(MONGODB_URI);
    console.log('数据库连接成功');

    // 检查是否已存在管理员账户
    const existingAdmin = await User.findOne({ username: 'admin' });
    if (existingAdmin) {
      console.log('管理员账户已存在，正在更新密码...');
      existingAdmin.password = 'admin123';
      await existingAdmin.save();
      console.log('管理员密码已更新');
    } else {
      // 创建新的管理员账户
      const admin = new User({
        username: 'admin',
        password: 'admin123',
        role: 'admin',
        status: 'active'
      });
      
      // 设置管理员权限
      admin.permissions = admin.getDefaultPermissions();
      await admin.save();
      console.log('管理员账户创建成功');
    }

    console.log('初始化完成');
    process.exit(0);
  } catch (error) {
    console.error('初始化失败:', error);
    process.exit(1);
  }
}

initAdmin(); 

File: backend/app/scripts/init_admin.py
from sqlalchemy.orm import Session
from app.models import User, UserRole, UserStatus
from app.core.security import get_password_hash
from app.database import SessionLocal

def init_superuser(db: Session) -> None:
    # 检查是否已存在超级管理员
    admin = db.query(User).filter(User.is_superuser == True).first()
    if admin:
        print("超级管理员已存在，跳过初始化")
        return

    # 创建超级管理员账号
    superuser = User(
        email="admin@any-go.com",
        username="admin",
        full_name="System Administrator",
        hashed_password=get_password_hash("admin123"),  # 请在首次登录后修改此密码
        is_active=True,
        is_superuser=True,
        role=UserRole.ADMIN,
        status=UserStatus.ACTIVE,
        permissions=[
            "products:read", "products:write", "products:delete",
            "packing_lists:read", "packing_lists:write", "packing_lists:delete",
            "users:read", "users:write", "users:delete",
            "system:backup"
        ]
    )

    try:
        db.add(superuser)
        db.commit()
        print("超级管理员账号创建成功")
        print("用户名: admin")
        print("密码: admin123")
        print("请在首次登录后立即修改密码！")
    except Exception as e:
        db.rollback()
        print(f"创建超级管理员失败: {str(e)}")

def main():
    db = SessionLocal()
    try:
        init_superuser(db)
    finally:
        db.close()

if __name__ == "__main__":
    main()